docs/book/src/reference/completion.md

```diff
@@ -0,0 +1,29 @@
+# Enable shell auto completion
+
+## Prerequisites
+
+- Bash 4.1+/zsh 
+- bash-completion/zsh-completions v2
+
+The kubebuilder completion script for bash can be generated with `kubebuilder completion <bash|zsh>`. Sourcing this script in your shell enables kubebuilder completion.
+
+- Once installed, go ahead and add the path `/usr/local/bin/bash` in the  `/etc/shells`.
+
+    `sudo echo “/usr/local/bin/bash” > /etc/shells`
```

Comment:
HI @bharathi-tenneti and @estroz,
  

- Why the sudo here is required? 
- Also, note that the bash requirements is just important for the bash one and not zsh. So, we cannot have the same pre-requirements for both. See my suggestion above
- Are these steps valid for any OS? So how can we add generic steps if they are not?. I think we could do similar the structure of the kubtecl docs here. See: https://kubernetes.io/docs/tasks/tools/install-kubectl/#enabling-shell-autocompletion

Reply:
Same concern as below https://github.com/kubernetes-sigs/kubebuilder/pull/1437#discussion_r400910399